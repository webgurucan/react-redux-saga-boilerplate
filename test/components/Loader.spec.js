import React from 'react';

import Loader from 'components/Loader';

function setup(type = 'pulse') {
  return mount(<Loader type={type} />);
}

describe('Loader', () => {
  let wrapper;

  it('should render properly with pulse type', () => {
    wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render properly with rotate type', () => {
    wrapper = setup('rotate');

    expect(wrapper).toMatchSnapshot();
  });
});
