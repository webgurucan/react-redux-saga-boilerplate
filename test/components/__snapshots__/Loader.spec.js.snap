// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render properly with pulse type 1`] = `
.c0 {
  height: 2rem;
  margin: 1.5rem auto;
  position: relative;
  width: 2rem;
}

.c0 > div {
  -webkit-animation: hmrJPG 1.15s infinite cubic-bezier(0.215,0.61,0.355,1);
  animation: hmrJPG 1.15s infinite cubic-bezier(0.215,0.61,0.355,1);
  border: 0 solid #00b4d5;
  border-radius: 50%;
  box-sizing: border-box;
  height: 0;
  left: 50%;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  width: 0;
}

<LoaderPulse>
  <div
    className="c0"
  >
    <div />
  </div>
</LoaderPulse>
`;

exports[`Loader should render properly with rotate type 1`] = `
.c0 {
  height: 10rem;
  margin: 1.5rem auto;
  width: 10rem;
}

.c1 {
  -webkit-animation: cZxgpV 2s linear infinite;
  animation: cZxgpV 2s linear infinite;
  height: 10rem;
  width: 10rem;
}

.c2 {
  -webkit-animation: drFgOE 1.5s ease-in-out infinite;
  animation: drFgOE 1.5s ease-in-out infinite;
  stroke: #333;
  stroke-dasharray: 1,200;
  stroke-dashoffset: 0;
  stroke-linecap: round;
  stroke-miterlimit: 10;
}

<LoaderRotate>
  <div
    className="c0"
  >
    <LoaderRotateSVG>
      <svg
        className="c1"
      >
        <LoaderRotateCircle
          cx="50"
          cy="50"
          fill="none"
          r="20"
          strokeWidth="2"
        >
          <circle
            className="c2"
            cx="50"
            cy="50"
            fill="none"
            r="20"
            strokeWidth="2"
          />
        </LoaderRotateCircle>
      </svg>
    </LoaderRotateSVG>
  </div>
</LoaderRotate>
`;
